### ISSUE TITLE
The `min-max-identity` does not trigger where expected due to limited type analysis

### ISSUE THREAD

**Issue by ruancomelli** [sourcery-ai/sourcery#321]
### Checklist

- [x] I have searched the [Sourcery documentation](https://docs.sourcery.ai/) for the issue, and found nothing
- [x] I have checked there are no open [bugs](https://github.com/sourcery-ai/sourcery/issues?q=label%3Abug) referencing the same bug or problem

### Description
The user Tortoise#3130 [reported on Discord](https://discord.com/channels/1035190801195597944/1035199416337961010/1070061537408073809) that the [`min-max-identity`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/min-max-identity/) refactoring is not triggering where expected, and I can reproduce the issue:

> Sourcery correctly refactors
> ```python
> if self.health < 0:
>     self.health = 0
> ```
> as
> ```python
> self.health = max(self.health, 0)
> ```
> 
> However,
> ```python
> if self.health > self.max_health:
>     self.health = self.max_health
> ```
> does **not** get refactored as:
> ```python
> self.health = min(self.health, self.max_health)
> ```

I investigated this issue and discovered that `min-max-identity` requires the compared value `self.max_health` to have a known, "simple" type, such as `int` or `float`. However, since our type inference engine is still not powerful enough to do class-level analyses, we cannot infer that `self.max_health` is an integer, and hence the refactoring is not suggested.

Simply annotating `self.max_health` is enough to get the refactoring to trigger:
```python
self.max_health: int
if self.health > self.max_health: # min-max-identity gets proposed!
    self.health = self.max_health
```
but this is far from ideal.

The expected behavior is to get the refactoring to trigger even without this additional type annotation.

### Potential solutions

There are two potential solutions for this bug:

1. **More powerful type analysis**: this is already in our roadmap but will still take some time to be implemented. A powerful type analysis engine would potentially be able to infer that `self.max_health` has a "simple type" in the snippet above;
2. **Lift the restriction on the type**: I don't see a strong reason to disable the refactoring for unknown types. Possibly there's room for lifting or at least relaxing the restriction on the operand types here that would allow this refactoring to trigger more often;

### Debug Information

IDE Version: 
VS Code 1.74.3

Sourcery Version: 
Sourcery v1.0.3

Operating system and Version:
Ubuntu 20.04.5 LTS

**Comment by ruancomelli** [comment]
A fix for this issue was just merged and will be available in the next Sourcery release :tada:

**Comment by Hellebore** [comment]
Released with 1.0.4
